openmdao/core/problem.py

```diff
@@ -220,10 +220,7 @@ def setup(self, check=True, out_stream=sys.stdout):
         # TODO: handle any automatic grouping of systems here...
 
         # divide MPI communicators among subsystems
-        if MPI:
-            self.root._setup_communicators(MPI.COMM_WORLD)
-        else:
-            self.root._setup_communicators(FakeComm())
+        self.root._setup_communicators(self._impl.world_comm())
```

Comment:
This might be a comment for bret, but should be really be using COMM_WORLD here? 
This will make us not play nice for a number of situations. Even nested problem instances could end up fowling this up.